pkg/logs/sender/sender.go

```diff
@@ -92,7 +97,9 @@ func (s *Sender) run() {
 			if !destSender.lastSendSucceeded {
 				if !destSender.NonBlockingSend(payload) {
 					tlmPayloadsDropped.Inc("true", strconv.Itoa(i))
+					telemetry.GetStatsTelemetryProvider().CountNoIndex(payloadsDroppedTelemetryMetricName, 1, []string{"reliable:true"})
 					tlmMessagesDropped.Add(float64(len(payload.Messages)), "true", strconv.Itoa(i))
+					telemetry.GetStatsTelemetryProvider().CountNoIndex(logsDroppedTelemetryMetricName, float64(len(payload.Messages)), []string{"reliable:true"})
```

Comment:
Just noticing that we seem to be using two separate patterns for emitting metrics. To satisfy my ignorance, when do you use `telemetry.GetStatsTelemetryProvider().X` vs. initializing a variable like `tlmPayloadsDropped`?

Reply:
`GetStatsTelemetryProvider` sends metrics directly to the intake - ex: users can query them in their org. 
agent (internal) telemetry is exposed as prometheus metrics and is not collected by default (requires an open metrics check to be setup - very verbose telemetry)

We expose a very limited subset of metrics via "stats telemetry" when they may be directly useful to users.

Reply:
Aha, gotcha, that distinction makes sense: whether the metrics are made immediately available to customers that upgrade vs. just published to the internal metrics store and available via the open metrics endpoint. In this case we think these metrics are high value enough to publish immediately? I'm assuming based on the analogous dropped points metric we have for metrics publishing?

Reply:
> In this case we think these metrics are high value enough to publish immediately? I'm assuming based on the analogous dropped points metric we have for metrics publishing?
Yep exactly

Reply:
We do have a facility to send prometheus metrics by default (e.g. used [here](https://github.com/datadog/datadog-agent/blob/cc21ca40c5c2f69312b5899ba5e6ff3bad912264/comp/forwarder/defaultforwarder/internal/retry/point_count_telemetry.go#L16)). Would it make sense to use it here?

Reply:
Thanks! Somehow I missed the PR that introduced this. Much better solution 👍